lib/server.go

```diff
@@ -634,14 +632,18 @@ func NewServer(
 
 	// Only initialize the FastHotStuffConsensus if the node is a validator with a BLS keystore
 	if _blsKeystore != nil {
+		snapshotGlobalParamsEntry, err := currentUtxoView.GetCurrentSnapshotGlobalParamsEntry()
+		if err != nil {
+			return nil, errors.Wrapf(err, "NewServer: Problem getting current snapshot global params entry"), true
+		}
 		srv.fastHotStuffConsensus = NewFastHotStuffConsensus(
 			_params,
 			srv.networkManager,
 			_chain,
 			_posMempool,
 			_blsKeystore.GetSigner(),
-			_posBlockProductionIntervalMilliseconds,
-			_posTimeoutBaseDurationMilliseconds,
+			snapshotGlobalParamsEntry.BlockProductionIntervalMilliseconds,
+			snapshotGlobalParamsEntry.TimeoutIntervalMilliseconds,
```

Comment:
Fetching the snapshotGlobalParams here can fetch stale snapshot global params if the chain is not synced yet.

Rather than passing the snapshot values here, it'll be more accurate to fetch the snapshotGlobalParams within the `FastHotStuffConsensus.Start(...)` function. The consensus only starts once the chain is sync'd so that makes sure it can't run with run or default values.